core/https-client-daemon/src/https_client.rs

```
@@ -0,0 +1,82 @@
+	See the License for the specific language governing permissions and
+	limitations under the License.
+
+*/
+#[cfg(all(not(feature = "std"), feature = "sgx"))]
+use crate::sgx_reexport_prelude::*;
+
+use crate::{
+	error::{Error, Result},
+	Request,
+};
+use itc_rest_client::{http_client::HttpClient, rest_client::RestClient, ResponseBody, RestGet};
+use itp_extrinsics_factory::CreateExtrinsics;
+use itp_ocall_api::EnclaveOnChainOCallApi;
+use itp_types::OpaqueCall;
+use log::*;
+use std::{string::String, time::Duration};
+use url::Url;
+
+const TIMEOUT: Duration = Duration::from_secs(3u64);
+
+/// Https rest client. Handles the https requests and responses.
+pub struct HttpsRestClient<T: EnclaveOnChainOCallApi, S: CreateExtrinsics> {
+	url: Url,
+	client: RestClient<HttpClient>,
+	ocall_api: T,
+	create_extrinsics: S,
+}
+
+impl<T: EnclaveOnChainOCallApi, S: CreateExtrinsics> HttpsRestClient<T, S> {
+	pub fn new(url: Url, ocall_api: T, create_extrinsics: S) -> Self {
+		let http_client = HttpClient::new(true, Some(TIMEOUT), None, None);
+		let rest_client = RestClient::new(http_client, url.clone());
+		Self { url, client: rest_client, ocall_api, create_extrinsics }
+	}
+
+	pub fn base_url(&self) -> &Url {
+		&self.url
+	}
+
+	/// Sends an https request to the specified server.
+	/// TODO: upon response create and send a parentchain extrinsic.
+	pub fn send(&mut self, request: Request) -> Result<()> {
+		let response = self
+			.client
+			.get::<String, ResponseBody>(request.request_str)
+			.map_err(|e| Error::Other(e.into()))?;
+
+		debug!("https get result as ( {:?},)", response);
+
+		let hardcode_score = 1234_u32;
+
+		let credit_score_module_id = 60;
+		let report_credit_score_method_id = 0;
+
+		let call = OpaqueCall::from_tuple(&(
+			[credit_score_module_id, report_credit_score_method_id],
+			request.account_id,
+			hardcode_score,
+		));
+		let calls = std::vec![call];
+
+		let tx = self.create_extrinsics.create_extrinsics(calls.as_slice()).unwrap();
```

Using unwrap here can cause panics, resulting in a shutdown of the current thread. I think that's not something that should ever happen. Can you return an error instead? (same for the unrwap below)